### PR TITLE
FEATURE: RAIL-894 Move menu to sidebar

### PR DESCRIPTION
**Pull request by czechian**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/192/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1432/ : FAILURE in 1m 27s

**Comment by czechian** [comment]
extended test

**Comment by czechian** [comment]
ok to test

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50015/storybook/

**Comment by czechian** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/194/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1433/ : SUCCESS in 3m 56s

**Comment by czechian** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/195/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/505/ : FAILURE in 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/197/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1435/ : FAILURE in 1m 08s

**Comment by czechian** [comment]
ok to test

**Comment by czechian** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/210/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1447/ : FAILURE in 4m 11s

**Comment by czechian** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/211/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1448/ : FAILURE in 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/212/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1449/ : SUCCESS in 1m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/213/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1450/ : SUCCESS in 1m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/214/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1451/ : SUCCESS in 1m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/215/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1452/ : SUCCESS in 1m 44s

File: examples/src/components/utils/Menu.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import PropTypes from 'prop-types';
+import { NavLink, withRouter } from 'react-router-dom';
+
+class Menu extends React.Component {
+    render() {
+        const { location: { pathname }, sideNavigationRoutes, routes, isUserLoggedIn } = this.props;
+        const href = pathname;
+        const currentRoute = (href !== undefined && routes.find(link => (link.path === BASEPATH + href))) || null;
+
+        if (!isUserLoggedIn) {
```
Comment (tomasreichmann) [inline examples/src/components/utils/Menu.jsx]: Do we really want to forbid users to go to the documentation for example if they are not logged in?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/217/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1454/ : SUCCESS in 1m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/218/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1455/ : SUCCESS in 1m 39s

**Comment by tomasreichmann** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/32/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1456/ : SUCCESS in 1m 50s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/601/ : SUCCESS in 16m 45s

**Pull request merged**